pleo-io/actions
# Fix RegExp  for lokalise-create-task

**This text is copied from https://github.com/pleo-io/product-web/pull/7875 which was used to test this change**

# Bug
`const GITHUB_HEADER_REGEX = /#{1,5} (.*)/g;` captures text from any line containing `#`, even if it's not an actual markdown header, such as URL fragment separator, as in the example below

# Solution
`const GITHUB_HEADER_REGEX = /^([\n]*[ ]*[#]{1,5})[ ]+(.*)$/gm;` will only match lines starting with '\n' or 'space' or nothing and have at least 1 space after the `#` character and at least 1 non-space character before the end of the line.
view the regExp here https://regex101.com/r/v8v5xk/1

# Discussion
By triggering the action with the examples below in this PR description, the original conditions are recreated for testing. The keys generated have been removed from Lokalise but the action runs can be found here:

1. example run before the fix:
https://github.com/pleo-io/product-web/actions/runs/2370638452%7C

2. run after changing the regex:
https://github.com/pleo-io/product-web/runs/6897174026?check_suite_focus=true

**TEST STRING. DO NOT TRANSLATE, PLS.**

# Other instance of the same bug as seen in https://github.com/pleo-io/product-web/runs/6859918194?check_suite_focus=true#step:9:62

### Run examples
  1. New strings, no `ready-for-translation` label on the PR:
  
  > [Runs as expected](https://github.com/pleo-io/product-web/runs/65086[56](https://github.com/pleo-io/product-web/runs/6859918194?check_suite_focus=true#step:9:56)688?check_suite_focus=true) and fails with a list of strings <img width="510" alt="Screenshot 2022-05-19 at 16 02 06" src="https://user-images.githubusercontent.com/103430887/169313909-b4aa8cda-0ff8-45f7-a47b-4bf1154cb509.png">
  
  4. New strings, with `ready-for-translation` label on the PR:
  
  > [Is skipped](#), as expected <img width="8[59](https://github.com/pleo-io/product-web/runs/6859918194?check_suite_focus=true#step:9:59)" alt="Screenshot 2022-05-19 at 16 19 12" src="https://user-images.githubusercontent.com/103430887/169316982-c331c5c3-6fce-4741-b516-b15b931838c2.png">
  
  5. No new strings, no `ready-for-translation` label on the PR:

# The body or the PR message causing the issue, as copied from https://github.com/pleo-io/product-web/runs/6553465933?check_suite_focus=true#step:7:9: 

  
  Linear Issue: [HEI-1769](https://linear.app/pleo/issue/HEI-1769/vat-step-add-name-of-the-company-in-fdd-flow)
  
  #### TO BE DONE
  
  ⚠️ 
  ⚠️ Send to localisation once approvals are in
  ⚠️ 
  
  ### CHANGES
  
  - Adds company name in the subtitle of VAT step in FDD flow as per design [here](https://linear.app/pleo/issue/HEI-1758#comment-1a0e893f)
  - Test
  
  ### HOW TO TEST
  
  1. [Log in](https://feature-hei-1769-vat-step-add-name-of-the-company-in-fdd.app.staging.pleo.io/login) to any company that has gone through signup but has not completed the company verification path yet.
  2. Reach VAT step - you should see the name of the company in the subtitle as per screenshot below.
  
  ### Quality checklist
  
  -   [x] Unit testing of logic updated/added
  -   [ ] Cypress UI tests for main user scenarios updated/added
  -   [ ] RTL tests for all user scenarios updated/added
  -   [ ] Storybook stories for distinct states of all screens updated/added
  -   [x] Manually tested the production version via the feature deployment app
  -   [ ] Requested translation with `ready-for-translation` label, or put `translation-exception` label for bypassing required translation for new messages
  -   [ ] [CODEOWNERS file](/pleo-io/product-web/blob/master/CODEOWNERS) updated
  
  ### Links to relevant storybook stories
  
  
  
  ### Information for translators
  
  
  
  
  
  
  
  ### Screenshots
  
  <img width="660" alt="Screenshot 2022-05-[11](https://github.com/pleo-io/product-web/runs/6553465933?check_suite_focus=true#step:7:11) at 11 56 16" src="https://user-images.githubusercontent.com/90062357/167835549-288f071a-77ba-457d-a99c-02d4150c406e.png">

---
**💻 App preview links**
_Latest_: https://feature-hms-440-fix-create-task.app.staging.pleo.io
_Current permalink_: https://preview-259a7b1cb3cc308c2c9a12bbeb12f0132201ea7b.app.staging.pleo.io


---
**📚 Storybook preview links**
_Latest_: https://feature-hms-440-fix-create-task.stories.staging.pleo.io
_Current permalink_: https://preview-259a7b1cb3cc308c2c9a12bbeb12f0132201ea7b.stories.staging.pleo.io
